controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +590,109 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS thingy
+			dnsRecord := &infranetworkv1.DNSData{}
+			dnsName := "dns-" + serviceName + fmt.Sprintf("-%d", i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			Log.Info(fmt.Sprintf("Name: %v, Namespace: %v", dnsName, dnsNamespace))
+			test_err := helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsRecord)
+			if test_err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", test_err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsRecord)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", err))
+			}
+
 		}
 	}
+
+	Log.Info("DNS Starts here")
+
+	serviceName = ovndbcluster.ServiceNameSB
+	if instance.Spec.DBType == v1beta1.NBDBType {
+		serviceName = ovndbcluster.ServiceNameNB
+	}
+	serviceLabels = map[string]string{
+		common.AppSelector: serviceName,
+	}
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if len(podList.Items) > int(*(instance.Spec.Replicas)) {
+
+	}
+	for i, ovnPod := range podList.Items {
+		if i >= int(*(instance.Spec.Replicas)) {
+			Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+			break
+		}
+		Log.Info(fmt.Sprintf("Using Pod: %v", ovnPod.Name))
+		var dnsName string
+		var dnsIP string
+		var hostnames []string
+		dnsName = "dns-" + ovnPod.Name
+		// Get Hostname
+		svc, _ := service.GetServiceWithName(
+			ctx,
+			helper,
+			ovnPod.Name,
+			ovnPod.Namespace,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+		hostnames = append(hostnames, hostname)
+
+		// Get IP
+		net_stat, _ := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+		for _, v := range net_stat {
+			if v.Interface == instance.Spec.NetworkAttachment {
+				dnsIP = v.IPs[0]
+			}
+		}
+
+		// Create DNSRecord
+		var DNSRecords []infranetworkv1.DNSHost
+		// ovsdbserver-sb-x entry
+		DNSRecord := infranetworkv1.DNSHost{}
+		DNSRecord.IP = dnsIP
+		DNSRecord.Hostnames = hostnames
+		// ovsdbserver-sb entry
+		headless_dns_hostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
```

Comment:
needed for both NB/SB so can use generic serviceName here so another if would not be needed., also .cluster.local suffix to be checked.